msmtools/analysis/dense/gpcca.py

```diff
@@ -1357,19 +1376,18 @@ def optimize(self, m, full_output=False):
             # coarse-grained stationary distribution
             self._pi_coarse = np.dot(self._chi.T, self._pi)
         except ValueError as err:
-            print("Stationary distribution couldn't be calculated:", err)
-                         
+            warnings.warn("Stationary distribution couldn't be calculated.")
+
         ## coarse-grained input (initial) distribution of states
         self._eta_coarse = np.dot(self._chi.T, self.eta)
 
         # coarse-grain transition matrix 
         self._P_coarse = coarsegrain(self.P, self.eta, self._chi)
-        
-        if full_output:
-            return (self, self._chi, self._rot_matrix, self._crispness, self.X, self.R, chi_list, rot_matrix_list, crispness_list)
-        else:
-            return (self, self._chi, self._rot_matrix, self._crispness, self.X, self.R)
-        
+
+        if return_extra:
+            return self, chi_list, rot_matrix_list, crispness_list
```

Comment:
self.X and self.R are not accessible via properties (only the reduced forms self._X and self._R are). If one would like to access them one would need to look into the source code. Thus I gave them back, if full_output was requested.